### PR TITLE
[PT1-606] Series epoch manager opcode

### PR DESCRIPTION
**Pull request by SteMak**
## Change Summary

Introducing `_validateSeriesEpochXD` opcode

The opcode allows order to be executed only if storage epoch matches the one specified in opcode parameter
The opcode supports series for single maker able to create independent order sequences

## Testing & Verification
**How was this tested?**
- [x] Unit tests
- [ ] Integration tests
- [ ] Manual testing (describe steps)
- [ ] Verified on staging

`forge test --match-path test/SeriesEpochManager.t.sol --fuzz-runs 100000`

## Risk Assessment
**Risk Level:**
- [ ] **Low** - Minor changes, no operational impact
- [x] **Medium** - Moderate changes, limited impact, standard rollback available
- [ ] **High** - Significant changes, potential operational impact, complex rollback

**Risks & Impact**

The PR introduces external functions to SwapVM Router: `seriesEpoch`, `seriesEpochAdvance`, `seriesEpochIncrease`, it is an open question if events are needed for such external functions

Opcode arguments note: at packing both series and epoch are provided as `uint32` while opcodes manage those as `uint256`, `uint32` is a soft-limit which can be extended to any reasonable bytes size

### CODE REVIEW

File: test/SeriesEpochManager.t.sol [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+            postTransferInHookData: "",
+            preTransferOutHookData: "",
+            postTransferOutHookData: "",
+            preTransferInCallbackData: "",
+            preTransferOutCallbackData: "",
+            instructionsArgs: "",
+            signature: ""
+        }));
+    }
+}
```
Comment (BoogerWooger) [inline test/SeriesEpochManager.t.sol:436]: Not all logic branches and boundary values are covered in tests. 
Suggest to add:

- boundary values for advance: amount = 1 and amount=255
- skipping large amount of epochs and make sure that all "intermediate" orders are cancelled
- testing "quote" for dead and live orders
- (maybe) handling epoch advance for at least two distinct makers -> advance first -> dead order for first, live order for second -> advance second -> both makers have dead orders

**Review by BoogerWooger — CHANGES_REQUESTED** [review]

File: test/SeriesEpochManager.t.sol (hunk 1 above)
Comment (SteMak) [reply test/SeriesEpochManager.t.sol:436]: added tests
- 0 -> revert
- boundary values for advance: amount = 1 and amount = 255 -> pass
- limited advance amount to uint8 -> no check for amount = 256 needed
- one more test for skipping multiple epochs and corresponding orders
- try execute validates consistency of quote and swap behavior
- check that different makers access different storage slots

**Review by ifelsedeveloper — APPROVED** [review]
looks good
